go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -32,6 +48,818 @@ func (c *Controller) processNextANPWorkItem(wg *sync.WaitGroup) bool {
+			return nil, err
+		}
+		acls = append(acls, acl...)
+	}
+
+	return acls, nil
+}
+
+// convertANPRuleToACL takes the given gressRule and converts it into an ACL(0 ports rule) or
+// multiple ACLs(ports are set) and returns those ACLs so that ops to add them to the pgName
+// can be constructed based on that.
+func (c *Controller) convertANPRuleToACL(rule *gressRule, pgName, anpName string) ([]*nbdb.ACL, error) {
+	// create address-set
+	// TODO (tssurya): Revisit this logic to see if its better to do one address-set per peer
+	// and join them with OR if that is more perf efficient. Had briefly discussed this OVN team
+	// We are not yet clear which is better since both have advantages and disadvantages.
+	// Decide this after doing some scale runs.
+	// TODO(tssurya): Use address-sets of namespaces whereever it makes sense; specially if namespace is already created
+	var err error
+	rule.addrSet, rule.podIPs, err = c.createASForPeers(rule.peers, rule.priority,
+		rule.gressPrefix, anpName, libovsdbops.AddressSetAdminNetworkPolicy)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create address set for "+
+			" rule %s with priority %d: %w", rule.name, rule.priority, err)
+	}
+	// create match based on direction and address-set
+	l3Match := constructMatchFromAddressSet(rule.gressPrefix, rule.addrSet)
+	// create match based on rule type (ingress/egress) and port-group
+	lportMatch := constructMatchFromPGName(rule.gressPrefix, pgName)
+	direction := getACLDirectionForRule(rule.gressPrefix)
+	options := getACLOptionsForRule(rule.gressPrefix)
+	var match string
+	var extIDs map[string]string
+
+	acls := []*nbdb.ACL{}
+	// no ports in this rule; so its a single ACL
+	if len(rule.ports) == 0 {
+		extIDs = getANPRuleACLDbIDs(anpName, rule.gressPrefix, fmt.Sprintf("%d", rule.priority),
+			controllerName, libovsdbops.ACLAdminNetworkPolicy).GetExternalIDs()
+		match = fmt.Sprintf("%s && %s", l3Match, lportMatch)
+		acl := libovsdbops.BuildACL(
+			getANPGressPolicyACLName(anpName, rule.gressPrefix, fmt.Sprintf("%d", rule.priority)),
+			direction,
+			int(rule.priority),
+			match,
+			getACLActionForANPRule(rule.action),
+			// TODO(tssurya): Hardcoding logging related parameters for now.
+			// Will fix this in future PRs when I add support for ANP-Logging
+			types.OvnACLLoggingMeter,
+			nbdb.ACLSeverityDebug,
+			false,
+			extIDs,
+			options,
+			types.DefaultANPACLTier,
+		)
+		acls = append(acls, acl)
+		return acls, nil
+	}
+
+	// we have ports defined in this rule, so its one ACL per port
+	// TODO(tssurya): There are some improvement PRs around this, adapt them in the future iterations
+	for i, port := range rule.ports {
+		extIDs = getANPRuleACLDbIDs(anpName, rule.gressPrefix, fmt.Sprintf("%d.%d", rule.priority, i),
+			controllerName, libovsdbops.ACLAdminNetworkPolicy).GetExternalIDs()
+		l4Match := constructMatchFromPorts(port)
+		match = fmt.Sprintf("%s && %s && %s", l3Match, lportMatch, l4Match)
+		acl := libovsdbops.BuildACL(
```

Comment:
yes I did want to talk to you about that.. I will add this to the `TODO` docs which is work in progress. This will be done as part of my ANP logging PR, the real value of ovn/acl is how it auto-populates all of that logging fields nicely... so I do intend to change this in the future.

Leaving this comment as unresolved so that its visible.

Reply:
added docs. resolving.

Reply:
maybe in that case we will leave empty name for acls, and decide on the format in that logging PR?

Reply:
hmm why do we want to leave the acl name empty? not sure I follow that part.. I think we can stick to the naming we have now (i can make changes to using index position instead of priority if that makes more sense) let's talk next week more on this..

Reply:
spoke with Nadia I am going to do a PR to cut out `ovn/acl.go` into its own package that both NP/EFW/ANP/BANP can use without any cycle dependency.

Reply:
https://github.com/ovn-org/ovn-kubernetes/pull/3797 - once that merges, I will reuse the utilities in that one

Reply:
this is done now, phew took a while to get everything setup, but I am now using most of the acl.go utils and that helped a lot  to delete some of the utils I had created....

Reply:
resolving comment.